src/Evolution/Systems/Cce/AnalyticSolutions/GaugeWave.cpp

```diff
@@ -0,0 +1,206 @@
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/SpinWeighted.hpp"
+#include "DataStructures/Tags.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/Variables.hpp"
+#include "Evolution/Systems/Cce/Tags.hpp"
+#include "NumericalAlgorithms/Spectral/SwshCollocation.hpp"
+#include "NumericalAlgorithms/Spectral/SwshInterpolation.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace Cce::Solutions {
+
+/// \cond
+GaugeWave::GaugeWave(const double extraction_radius, const double mass,
+                     const double frequency, const double amplitude,
+                     const double peak_time, const double duration) noexcept
+    : SphericalMetricData{extraction_radius},
+      mass_{mass},
+      frequency_{frequency},
+      amplitude_{amplitude},
+      peak_time_{peak_time},
+      duration_{duration} {}
+
+std::unique_ptr<WorldtubeData> GaugeWave::get_clone() const
+    noexcept {
+  return std::make_unique<GaugeWave>(*this);
+}
+
+double GaugeWave::coordinate_wave_function(const double time) const noexcept {
+  const auto retarded_time = time - extraction_radius_;
+  return amplitude_ * sin(frequency_ * retarded_time) *
+         exp(-square(retarded_time - peak_time_) / square(duration_));
+}
+
+double GaugeWave::du_coordinate_wave_function(const double time) const
+    noexcept {
+  const auto retarded_time = time - extraction_radius_;
+  return amplitude_ *
+         (-2.0 * (retarded_time - peak_time_) / square(duration_) *
+              sin(frequency_ * retarded_time) +
+          frequency_ * cos(frequency_ * retarded_time)) *
+         exp(-square(retarded_time - peak_time_) / square(duration_));
+}
+
+double GaugeWave::du_du_coordinate_wave_function(const double time) const
+    noexcept {
+  const auto retarded_time = time - extraction_radius_;
+  return amplitude_ *
+         (-4.0 * square(duration_) * (retarded_time - peak_time_) * frequency_ *
+              cos(frequency_ * retarded_time) +
+          (-2.0 * frequency_ * square(duration_) +
+           4.0 * square(retarded_time - peak_time_) -
+           pow<4>(duration_) * square(frequency_)) *
+              sin(frequency_ * retarded_time)) /
+         pow<4>(duration_) *
+         exp(-square(retarded_time - peak_time_) / square(duration_));
+}
+
+void GaugeWave::spherical_metric(
+    const gsl::not_null<
+        tnsr::aa<DataVector, 3, ::Frame::Spherical<::Frame::Inertial>>*>
+        spherical_metric,
+    const size_t /*l_max*/, const double time) const noexcept {
+  const auto wave_f = coordinate_wave_function(time);
+  const auto du_wave_f = du_coordinate_wave_function(time);
+  get<0, 0>(*spherical_metric) = -(extraction_radius_ - 2.0 * mass_) *
+                                 square(extraction_radius_ + du_wave_f) /
+                                 pow<3>(extraction_radius_);
+  get<0, 1>(*spherical_metric) =
```

Comment:
this is just for my own clarification, but since this is a tnsr::aa object I assume that it's forced to be symmetric?

Reply:
Yeah, that's right -- in fact, in spectre the symmetric tensors only have (N^2 + N)/2 storage slots in memory, so the `get<0, 1>(tensor)` points to literally the same data as `get<1, 0>(tensor)`.